homeassistant/components/generic/config_flow.py

```diff
@@ -0,0 +1,162 @@
+"""Config flow for generic (IP Camera)."""
+
+import imghdr
+import logging
+from typing import Any, Dict
+
+import av
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.camera import DEFAULT_CONTENT_TYPE
+from homeassistant.const import (
+    CONF_AUTHENTICATION,
+    CONF_NAME,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    CONF_VERIFY_SSL,
+    HTTP_BASIC_AUTHENTICATION,
+    HTTP_DIGEST_AUTHENTICATION,
+)
+
+from .camera import GenericCamera
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_CONTENT_TYPE,
+    CONF_FRAMERATE,
+    CONF_LIMIT_REFETCH_TO_URL_CHANGE,
+    CONF_RTSP_TRANSPORT,
+    CONF_STILL_IMAGE_URL,
+    CONF_STREAM_SOURCE,
+    DEFAULT_NAME,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class GenericIPCamConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for generic IP camera."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialise the config flow."""
+        super().__init__()
+        self._errors = {}
```

Comment:
The config flow design doesn't allow passing errors between steps.  While that isn't happening in this PR, setting this here may encourage it to happen in a future PR so please don't.

Reply:
Agree. Done.